### PR TITLE
docs: add initial version of aux table mdbook docs

### PR DESCRIPTION
**Pull request by grjte**
This is a rough overview of the current design of the Auxiliary Table's execution trace which outlines the considerations of the current implementation and notes some other options for the future.

Once v0.2 is finished and the design is more settled this should be revisited and updated.

### CODE REVIEW

File: docs/src/design/aux_table/main.md [hunk 1]
```diff
@@ -1,3 +1,184 @@
...
+Each co-processor is identified by a set of selector columns which identify its segment in the auxiliary table and cause its constraints to be selectively applied.
+
+During the finalization of the overall execution trace, each co-processor's trace and selector columns are appended to the Auxiliary Table one after another, such that when one co-processor trace ends the trace of the next co-processor starts in the subsequent row.
+
+Additionally, a padding segment must be added to the Auxiliary Table so that the number of rows in the table always matches the overall trace length of the other VM processors, regardless of the length of the co-processor traces. The padding will simply contain zeroes.
+
+## Design Requirements
+
+- The minimum width of the Auxiliary Table is 18 columns, which will fit the Hash processor and 1 selector column to select it.
+- The maximum number of constraint degrees for the VM is 9.
```
Comment (bobbinth) [inline docs/src/design/aux_table/main.md:24]: nit: this is slightly confusing. maybe say "The maximum constraint degree for the VM is 9."?

File: docs/src/design/aux_table/main.md [hunk 2]
```diff
@@ -1,3 +1,184 @@
...
+
+### Hasher
+
+- The hasher's constraints are already degree 8, so we are restricted to a 1-degree selector flag.
+- The hasher requires 17 columns, so its selector flag must require a single column to keep our aux table to the minimum.
+- Each hash operation in the hash processor is performed in an 8-row cycle that must begin at a row number equal to $0 mod8$, so the hash processor's trace must begin on a row number equal to $0 mod8$.
```
Comment (bobbinth) [inline docs/src/design/aux_table/main.md:30]: nit: I'm not sure if "$0 mod8$" will render correctly. should it be `$0 \mod 8$`? or maybe `$0\ mod\ 8$`?

File: docs/src/design/aux_table/main.md [hunk 3]
```diff
@@ -1,3 +1,184 @@
...
+- Let Processor A, Processor B, and Processor C be three co-processors whose execution traces are stacked such that they share the same columns, the trace rows for B start immediately after the trace rows for A end, and the trace rows for C start immediately after the end of the trace rows for B.
+- Let $s_0$ and $s_1$ be selector columns. When $s_0 = 0$, we apply the constraints for Processor A. When $s_0 = 1$ and $s_1 = 0$ we apply the constraints for Processor B. When $s_0 = 1$ and $s_1$ = 1 we apply the constraints for Processor C.
+
+Multiplying Processor A's constraints by $not(s_0')$ instead of by $not(s_0)$ will cause them to stop being enforced after the second-to-last row. For Processor B, this flag would look like $s_0 * not(s_1')$.
+
+This resolves our last-row issue, but it introduces a problem with the first row of a processor's trace.
+
+- If this construction is used for the first processor in the auxiliary table, then the very first row won't be constrained properly unless a validity constraint is defined for that specific row.
+- When used for subsequent processors, the following scenario is possible:
+  s0 = 0 …Processor A…
```
Comment (bobbinth) [inline docs/src/design/aux_table/main.md:168]: nit: should `s0` be written using math notation - e.g., `$s_0$`?

**Review by bobbinth — APPROVED** [review]
Looks great! Thank you! A couple of minor nits inline.

File: docs/src/design/aux_table/main.md (hunk 2 above)
Comment (grjte) [reply docs/src/design/aux_table/main.md]: Updated. I think it doesn't look great either way, actually, but this is more correct.

**Pull request merged**